### ISSUE TITLE
demand doesn't effect result properly

### ISSUE THREAD

**Issue by ilkot** [HamidM6/croston#1]
Hello,
Thanks for the package.
While I run some tests I realized that forecast doesn't change when demand changes.
When I run croston-original(h=1) for below example last demand's(last element of x-array which is 4) forecast is 4.63
also horizon=1(I added it as 'forecast') has same amount =4.63 
I guess it should change for step ahead when demand occurs, is it correct? or how does it work?

![croston_issuye](https://user-images.githubusercontent.com/38972709/84129015-0962e780-aa4a-11ea-9a2b-8cd5b99e9f3e.JPG)

```
import pandas as pd
import numpy as np
from croston import croston

x = np.array([0,0,0,10,0,4,0,0,5,0,0,0,2,10,0,20,2,18,0,0,14,4,12,0,0,0,3,1,0,16,4])
fit_pred_croston_sample_2 = croston.fit_croston(x,1,'original')
yhat_croston_sample_2 = pd.DataFrame(np.concatenate([fit_pred_croston_sample_2['croston_fittedvalues'], fit_pred_croston_sample_2['croston_forecast']]))
x = pd.DataFrame(np.hstack((x,np.array(['forecast']))))
result_sample = pd.merge(x, yhat_croston_sample_2, left_index=True, right_index=True)
```

**Comment by HamidM6** [comment]
Last fitted value is the future forecast regardless of forecast horizon.

**Comment by ilkot** [comment]
Thanks for the comment.
If I calculate the error for actual demand between forecast it is misleading, right?
So, I should shift all forecat results by 1 in order to calculate correctly, right?

**Comment by HamidM6** [comment]
I don't follow. You might want to ask questions in [stack exchange](https://stackexchange.com/). I have found it helpful.

**Comment by ilkot** [comment]
Thanks for the suggestion but I don't think it is a general Croston method question.
I agree with you about last fitted value is the forecast but if I compare what I forecasted before and what happened in reality doesn't match in row-wise.  It seems like a bug.

**Comment by HamidM6** [comment]
could you post an example please?

**Comment by HamidM6** [comment]
this is your last example and it looks good to me.

`
import pandas as pd   
import numpy as np   
from croston import croston   

y = np.array([0,0,0,10,0,4,0,0,5,0,0,0,2,10,0,20,2,18,0,0,14,4,12,0,0,0,3,1,0,16,4])    

fit_fcst = croston.fit_croston(y,1,'original')    

y_fitted = fit_fcst['croston_fittedvalues']    

(len(y), len(y_fitted))     
`

**Comment by ilkot** [comment]
Yes but forecasted value and last fitted value is same. It shouldn't be since there is a demand in last period which is 4. (bold in below)
y = np.array([0,0,0,10,0,4,0,0,5,0,0,0,2,10,0,20,2,18,0,0,14,4,12,0,0,0,3,1,0,16,**4**])

Below code produces same number but I guess value of y_fitted[-2] should be last fitted value.

```
y_fitted[-1]
Out[34]: 4.632864052702323

fit_fcst['croston_forecast']
Out[35]: array([4.63286405])
```

**Comment by HamidM6** [comment]
i don't think so.
you can look at these examples:

[https://otexts.com/fpp2/simple-methods.html](url)

**Comment by ilkot** [comment]
I think I couldn't explain myself clearly. Sorry about that.
I want to try what's on my mind in other way.

In below screenshot left side represents actual demands(y) and right side is fitted values or forecasts for corresponding demand(y_fitted). I think this assumption is correct.
![image](https://user-images.githubusercontent.com/38972709/84325783-39160a80-ab84-11ea-82db-d02a3f879ef3.png)

Let's say **t** is the observation number or month and when t = 0(January) , demand = 0, forecast = 0(first row) it is clear and expected.  
But when t = 3(April),  demand = 10(first demand in that series) and forecast =5 which is odd because there is no way to calculate it since there is no demand before that. 
Forecast = 5 has to be calculated in March( when t=2) for next month(April), but since demand in all zeros before April forecast would be 0 as well. 
In my opinion calculation is correct but need to be shifted by 1.
For example;
When t = 3, demand = 10, forecast = 0
When t = 4, demand = 0,   forecast = 5
When t = 5, demand = 4,   forecast = 5
When t = 6, demand = 0,   forecast = 3.8307 and so on.

If not please let me know how to calculate when t = 3, forecast = 5 without using demand=10
Thanks again.

**Comment by HamidM6** [comment]
This is a great question for stack exchange with plenty of stat experts that can help. Please post the question link here so others can follow.
Thank you for your time an effort.

**Comment by ilkot** [comment]
I posted the question on stack exchange as you suggest.

https://stats.stackexchange.com/questions/471590/croston-method-with-python-demand-doesnt-effect-forecast-properly


Thanks,
